pywbemtools/pywbemcli/pywbemcli.py

```diff
@@ -374,34 +373,41 @@ def create_server_instance(connection_name):
-                        "connections file '{}'".
-                        format(connection_name,
-                               connections_repo.connections_file))
+                        'Connection name: "{}" not found in connections file: '
+                        '"{}"'.format(connection_name,
+                                      connections_repo.connections_file))
 
             # No connection name specified, try default connection name
             else:
                 # get the default_connection definition
-                if connections_repo.file_exists():
-                    connection_name = \
-                        connections_repo.get_default_connection_name()
-
-                    # Abort because file modified to clear the bad default.
-                    # This should never happen andif it does, Abort is logical.
-                    if connection_name and connection_name not in \
-                            connections_repo:
-                        connections_repo.set_default_connection(None)
-                        click.echo(
-                            "Default connection definition '{}' not found in "
-                            "connections file '{}'; deleted default connection "
-                            "in connections file".
-                            format(connection_name,
-                                   connections_repo.connections_file),
-                            err=True)
-                        raise click.Abort()
-                    if verbose:
-                        click.echo('Current connection: "{}"'
-                                   .format(connection_name))
+                try:
+                    if connections_repo.file_exists():
+                        connection_name = \
+                            connections_repo.default_connection_name
+
+                        # Test if the default connection name is actually
+                        # in the repo.  If not reset the default connection
+                        # name to None and continue.
+                        # This should never occur unless connection file
+                        # is corrupted.
+                        if connection_name and connection_name not in \
+                                connections_repo:
+                            connections_repo.default_connection_name = None
+                            click.ClickException(
+                                'Default connection name: "{}" not found in '
+                                'connections file "{}"; default connection '
+                                'name is cleared.'.
+                                format(connection_name,
+                                       connections_repo.connections_file))
+                        if verbose:
+                            click.echo('Current connection: "{}"'
+                                       .format(connection_name))
+                except Exception as exc:
```

Comment:
I don't like broad exception catching around a large chunk of code. The only exception this code can possibly raise is ClickException.

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
DONE, No more except